apps/www/src/state/plugin/view.js

```diff
@@ -27,16 +31,28 @@ export const viewForm = {
       setSaveOperation(state, saveOperation) {
         return { ...state, saveOperation }
       },
-      setSourcesValidity(state, report) {
-        return { ...state, sourcesValidity: report }
+      setSourcesValidity(state, { report, shapes, pointer }) {
+        return { ...state, sourcesValidity: { report, shapes, pointer } }
       },
-      setViewValidity(state, validityReport) {
-        return { ...state, validityReport }
+      setViewValidity(state, { report, shapes, pointer }) {
+        return { ...state, validity: { report, shapes, pointer } }
       },
     },
     effects(store) {
       const dispatch = store.getDispatch()
 
+      function whenValid(effect) {
```

Comment:
perhaps 'checkValidity' or 'whenValidity' ? as it could be not valid...

Reply:
The intended semantics is "when valid, call `effect`".

Reply:
then what about 'whenValidCall' ?